src/HasTranslatableSlug.php

```diff
@@ -52,13 +53,50 @@ protected function addSlug()
         });
     }
 
+    protected function generateNonUniqueSlug(): string
+    {
+        $slugField = $this->slugOptions->slugField;
+        $slugString = $this->getSlugSourceString();
+
+        $originalSlug = $this->getOriginal($slugField)[$this->getLocale()] ?? null;
+        $newSlug = $this->getTranslations($slugField)[$this->getLocale()] ?? null;
+
+        $hasCustomSlug = $this->hasCustomSlugBeenUsed() && !empty($newSlug);
+        $hasChangedCustomSlug = $slugString == $this->getOriginalSourceString() && !is_null($newSlug) && $originalSlug == $newSlug;
```

Comment:
I don't understand this. Could you refactor this statement to a dedicated function that doesn't use any composite conditionals?

Reply:
Yeah, this could be better. I changed the readability. 

It's to prevent a custom slug from another locale get overwritten when another slug is changed. 
See this test: https://github.com/spatie/laravel-sluggable/pull/190/files#diff-a7c0d68fd2903431107578b05dbecb899480791e8ce134a279ee6db08e0ff128R201

So the issue was when you have a custom slug for e.g. Dutch and English, and you change only the English version, the Dutch version was also changed back to the title version.